server/src/main/scala/ch.epfl.scala.index.server/PublishProcess.scala

```diff
@@ -0,0 +1,200 @@
+package ch.epfl.scala.index
+package server
+
+import data.bintray._
+import data.cleanup.GithubRepoExtractor
+import data.download.PlayWsDownloader
+import data.elastic._
+import data.github.{GithubCredentials, GithubDownload}
+import data.maven.PomsReader
+import data.project.ProjectConvert
+
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Path}
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+
+import org.joda.time.DateTime
+
+import play.api.libs.ws.WSAuthScheme
+
+import com.sksamuel.elastic4s._
+import ElasticDsl._
+
+import scala.concurrent.{Await, Future}
+
+class PublishProcess(
+  implicit val system: ActorSystem,
+  implicit val materializer: ActorMaterializer,
+  implicit val api: ApiImplementation
+) extends PlayWsDownloader {
+
+  import system.dispatcher
+
+  /**
+   * resolve the filename for a specific pom by sha1
+   *
+   * @param sha1 the sha1 hash of the file
+   * @return
+   */
+  private def pomPath(sha1: String) = bintrayPomBase.resolve(s"$sha1.pom")
+  private def tmpPath(sha1: String) = tmpBase.resolve(s"$sha1.pom")
+
+  private def computeSha1(data: String): String = {
+
+    val md = java.security.MessageDigest.getInstance("SHA-1")
+    md.digest(data.getBytes("UTF-8")).map("%02x".format(_)).mkString
+  }
+
+  /**
+   * verify the sha1 file hash
+   *
+   * @param path the path to the file to verify
+   * @param sha1 the sha1 hash
+   * @return
+   */
+  private def verifySHA1FileHash(path: Path, sha1: String): Boolean = {
+
+    val source = scala.io.Source.fromFile(path.toFile)
+    val content = source.mkString
+    source.close()
+
+    verifyChecksum(content, sha1)
+  }
+  /**
+   * will compute ans sha1 hash from given string and verify that against
+   * given compare hash
+   *
+   * @param toVerify the string to compute
+   * @param sha1 the verify hash
+   * @return
+   */
+  def verifyChecksum(toVerify: String, sha1: String) = {
+
+    computeSha1(toVerify) == sha1
+  }
+
+  def writeFiles(fileName: String, data: String, credentials: GithubCredentials) = Future {
+
+
+    val tmpName = fileName.replace(".sha1", "").replace(".pom", "")
+    //println(s"Receive: $tmpName")
+    val hash = computeSha1(tmpName)
+    val path = tmpPath(hash)
+
+    if (fileName matches """.*\.pom.sha1""") {
```

Comment:
@MasseGuillaume Du you have an idea how to simplify that, maybe we can tell sbt to only publish *.pom and *.pom.sha1